core/src/main/java/com/es/core/model/phone/JdbcPhoneDao.java

```diff
@@ -1,27 +1,73 @@
 package com.es.core.model.phone;
 
+
+import com.es.core.enumeration.SortField;
+import com.es.core.enumeration.SortOrder;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.springframework.jdbc.core.BeanPropertyRowMapper;
 import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
 import org.springframework.stereotype.Component;
 
 import javax.annotation.Resource;
-import java.util.List;
-import java.util.Optional;
+import java.util.*;
 
 @Component
-public class JdbcPhoneDao implements PhoneDao{
+public class JdbcPhoneDao implements PhoneDao {
+    private static final String SQL_GET_COLORS_QUERY = "SELECT colors.id, colors.code FROM colors JOIN phone2color ON colors.id = phone2color.colorId WHERE phoneId = %d";
+    private static final String SQL_GET_PHONE_BY_ID_QUERY = "SELECT * FROM phones WHERE id = %d";
+    private static final String SQL_ALL_PRODUCTS = "SELECT * FROM phones ORDER BY %s %s OFFSET %d LIMIT %d";
+    private static final String SQL_FIND_PRODUCTS_NONNULL_PRICE_POSITIVE_STOCK = "SELECT * FROM phones JOIN stocks ON phones.id = stocks.phoneId WHERE price IS NOT NULL AND stock > 0 AND LOWER(model) LIKE LOWER(?) ORDER BY %s %s OFFSET %d LIMIT %d";
```

Comment:
It works when you specify LOWER('%%%s%%') - don't forget the quotes, as jdbcTemplate no longer inserts them for you when using String#format way. However, I think I'd personally stick to JDBC-args way everywhere in the class, it's a clearer use of the provided API than manually constructing SQL strings in my opinion. Perhaps, you can even try to look at [NamedParameterJdbcTemplate](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/jdbc/core/namedparam/NamedParameterJdbcTemplate.html) to make the code even more pretty and readable.